flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1620,6 +1620,16 @@ def test_created_volume_attributes(self):
             (new_volume.dataset_id, new_volume.size)
         )
 
+    def unknown_blockdevice_id(self):
+        """
+        Return a legitimate unknown block device id.
+
+        Override in subclasses as needed.
+
+        :return: By default a newly generated UUID.
+        """
+        return unicode(uuid4())
```

Comment:
Maybe better not to have a default implementation at all.  It's easy to overlook and might grant the appearance of correctness if malformed identifiers trigger different behavior from well-formed identifiers not associated with a volume.  `raise NotImplementedError(...)` would probably be good.